### ISSUE TITLE
[Bug][DeepSeek V4] Compressor SetBaseSize keeps large mBaseSize for varlen batches

### ISSUE THREAD

**Issue by QwertyJack** [vllm-project/vllm-ascend#9496]
## Summary

`aclnnCompressor` can produce different compressed KV results for the same active request when the request is served alone vs inside an uneven batched prefill. The issue was isolated to `CompressorKernelPerf<COMP>::SetBaseSize()` in:

`csrc/attention/compressor/op_kernel/compressor_kernel_perf.h`

The current logic only shrinks `constInfo.mBaseSize` when all batch elements have the same used sequence length. For uneven batches, for example `cu_seqlens=[0, 12, 23]`, it keeps the default `mBaseSize=128` even though the actual small-M workload should use a smaller base size.

## Impact

This affects DeepSeek V4 W4A8 serving on Ascend in batched prefill. In the local real-weight investigation, the same active request had identical compressor inputs between single-request and batched-request runs, but the compressor output differed by a few BF16 elements. The difference was then written into compressed KV cache and propagated to later attention/logits.

This is not a tokenizer, sampler, API server, graph-mode, FlashComm, FUSED_MC2, or MoE routing issue for this specific symptom: the first observed divergence was inside `_C_ascend.compressor` with identical active inputs.

## Evidence from local isolation

For the affected request, the following compressor inputs were equal between single and batched execution:

- `hidden_states`
- `wkv_weight`
- `wgate_weight`
- `ape`
- `norm_weight`
- `rope_sin` / `rope_cos`
- compact `state_cache_before`

The old compressor output differed in `compressed_kv` for 3 BF16 elements, with max absolute difference `0.00048828125`.

After changing `SetBaseSize()` to use the maximum used sequence length across the varlen batch, the offline replay result became exact for the same active rows:

- `single_vs_batch_code_active3 diff_count=0`
- `batch_req0_as_single_vs_single_code_active3 diff_count=0`

The fix was also validated in a TP16DP2EP real-weight smoke run: the single request and the same request inside a mixed batch returned the same first token and logprob.

## Root cause

The old logic effectively did this:

```cpp
bool sameSeqUsed = true;
uint32_t firstBatchSeqUsed = tools_.GetSeqLength(0);
...
if (sameSeqUsed && (bSeqUsed != firstBatchSeqUsed)) {
    sameSeqUsed = false;
}
...
if (sameSeqUsed && mSize <= (...)) {
    ...
    uint32_t newMBaseSize = ... * firstBatchSeqUsed;
    ...
}
```

For uneven varlen batches, `sameSeqUsed` becomes false, so small-M base-size adjustment is skipped completely. That leaves the default `mBaseSize=128`, which was shown to produce a different compressor result for the same active request.

## Suggested fix

Remove the equal-length gate and compute the candidate base size from the maximum used sequence length in the batch:

```cpp
uint32_t firstBatchSeqUsed = tools_.GetSeqLength(0);
uint32_t maxBatchSeqUsed = firstBatchSeqUsed;
for (uint32_t i = 0; i < constInfo.batchSize; i++) {
    uint32_t bSeqUsed = tools_.GetSeqLength(i);
    ...
    if (bSeqUsed > maxBatchSeqUsed) {
        maxBatchSeqUsed = bSeqUsed;
    }
}
...
if (mSize <= (constInfo.mBaseSize * (aiCoreNum / dBaseBlockNum))) {
    ...
    uint32_t newMBaseSize =
        (constInfo.batchSize + coreGroupNum - 1) / coreGroupNum * maxBatchSeqUsed;
    if (newMBaseSize > minMBaseSize && newMBaseSize < constInfo.mBaseSize) {
        constInfo.mBaseSize = newMBaseSize;
    }
}
```

## Notes

This issue is scoped to the compressor varlen-batch drift. There is a separate DeepSeek V4 MoE ALLTOALL fallback investigation around TP padding rows and active-row metadata; that should be tracked separately.

**Comment by weiguihua2** [comment]
@Nagisa125 please take a look

**Comment by Nagisa125** [comment]
@GDzhu01麻烦看一下
